### ISSUE TITLE
PREDICTABLE RANDOM NUMBER GENERATOR [Security issue]

### ISSUE THREAD

**Issue by lhuria94** [tradle/react-native-crypto#57]
Issue:
The mobile application uses a predictable Random Number Generator (RNG).
Under certain conditions this weakness may jeopardize mobile application data encryption or other protection based on randomization. For example,
if encryption tokens are generated inside of the application and an attacker can provide application with a predictable token to validate and then
execute a sensitive activity within the application or its backend.

Reference:
- https://developer.apple.com/library/content/documentation/Security/Conceptual/cryptoservices/GeneralPurposeCrypto/GeneralPurposeCrypto.html
- https://developer.apple.com/library/content/documentation/Security/Conceptual/cryptoservices/RandomNumberGenerationAPIs/RandomNumberGenerationAPIs.html

Can you help if this is related to crypto randomBytes function?

**Comment by mvayngrib** [comment]
@lhuria94 what makes u say it uses a predictable random number generator?

**Comment by lhuria94** [comment]
I am not saying, its the scan by the security team. 

Here is the example reference they provided:
![Screenshot 2020-06-10 at 5 48 57 PM](https://user-images.githubusercontent.com/10358379/84266791-b15bdc00-ab42-11ea-9eb6-62dbaea01f5f.png)

![Screenshot 2020-06-10 at 5 49 40 PM](https://user-images.githubusercontent.com/10358379/84266844-c6d10600-ab42-11ea-8a24-f0488e809102.png)

With details like:
* Binary match usage of 'random' function/method.
• Binary match usage of 'srand' function/method.

**Comment by mvayngrib** [comment]
this it the randombytes module used: https://github.com/mvayngrib/react-native-randombytes

the method of getting random bytes depends on async/sync usage: https://github.com/mvayngrib/react-native-randombytes#usage

the async method: https://github.com/mvayngrib/react-native-randombytes/blob/master/RNRandomBytes.m#L32

the sync method uses SJCL with a random seed generated by SecRandomCopyBytes

**Comment by lhuria94** [comment]
Oh okay, that means instead of using it like:
`const password = crypto.randomBytes(10).toString('hex');`

We should use:
```
// asynchronous API
// uses iOS-side SecRandomCopyBytes
const password = randomBytes(10, (err, bytes) => {
  // bytes is a Buffer object
  return bytes.toString('hex'));
})
```

**Comment by mvayngrib** [comment]
@lhuria94 yes, that would be the more secure way to use it. I've been meaning to update react-native-randombytes to use the newly available synchronous react-native bridge methods (via RCT_EXPORT_SYNCHRONOUS_TYPED_METHOD), so that the sync method would work as well but haven't gotten to it yet

**Comment by lhuria94** [comment]
No worries, Thanks a lot for the quick help. Ill update if this resolves the security issue.
